False positive `invalid-enum-extension` for Enum parent class with annotations

### Bug description

Pylint incorrectly flags an Enum inherited from a parent Enum which has annotations. This is useful when overriding `__new__` or `__init__` on the parent Enum to add extra data to the Enum members while also keeping the codebase typed. Example:

`example.py`
```
from enum import Enum


class ColorEnum(Enum):
    red: int
    green: int
    blue: int

    def __init__(self, red: int, green: int, blue: int) -> None:
        self.red = red
        self.green = green
        self.blue = blue

class Pastel(ColorEnum):
    SAGE = (170, 200, 167)
```

### Configuration

_No response_

### Command used

```shell
pylint example.py
```


### Pylint output

```shell
************* Module example
example.py:14:0: E0244: Extending inherited Enum class "ColorEnum" (invalid-enum-extension)
```


### Expected behavior

No output

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.8.16 (default, Mar 15 2023, 13:36:46)
[Clang 14.0.0 (clang-1400.0.29.202)]
```


### OS / Environment

macOS Ventura 13.2.1

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/8830.false_positive
+++ b/doc/whatsnew/fragments/8830.false_positive
@@ -1,0 +1,4 @@
+Exempt parents with only type annotations from the ``invalid-enum-extension``
+message.
+
+Closes #8830

--- a/pylint/checkers/classes/class_checker.py
+++ b/pylint/checkers/classes/class_checker.py
@@ -893,12 +893,21 @@ a metaclass class method.",
     def _check_enum_base(self, node: nodes.ClassDef, ancestor: nodes.ClassDef) -> None:
         members = ancestor.getattr("__members__")
         if members and isinstance(members[0], nodes.Dict) and members[0].items:
-            self.add_message(
-                "invalid-enum-extension",
-                args=ancestor.name,
-                node=node,
-                confidence=INFERENCE,
-            )
+            for _, name_node in members[0].items:
+                # Exempt type annotations without value assignments
+                if all(
+                    isinstance(item.parent, nodes.AnnAssign)
+                    and item.parent.value is None
+                    for item in ancestor.getattr(name_node.name)
+                ):
+                    continue
+                self.add_message(
+                    "invalid-enum-extension",
+                    args=ancestor.name,
+                    node=node,
+                    confidence=INFERENCE,
+                )
+                break
 
         if ancestor.is_subtype_of("enum.IntFlag"):
             # Collect integer flag assignments present on the class

--- a/tests/functional/i/invalid/invalid_enum_extension.py
+++ b/tests/functional/i/invalid/invalid_enum_extension.py
@@ -21,6 +21,33 @@ class D(C):
     x = 3
 
 
+# Similarly, items that are only type annotations are okay.
+class ColorEnum(Enum):
+    red: int
+    green: int
+    blue: int
+
+    def __init__(self, red: int, green: int, blue: int) -> None:
+        self.red = red
+        self.green = green
+        self.blue = blue
+
+
+class Pastel(ColorEnum):
+    SAGE = (170, 200, 167)
+
+
+class IncorrectColorEnum(Enum):
+    red: None = None
+
+    def __init__(self, red: None) -> None:
+        self.red = red
+
+
+class IncorrectPastel(IncorrectColorEnum):  # [invalid-enum-extension]
+    SOME_COLOR = 170
+
+
 class CustomFlags(IntFlag):
     SUPPORT_OPEN = 1
     SUPPORT_CLOSE = 2

--- a/tests/functional/i/invalid/invalid_enum_extension.txt
+++ b/tests/functional/i/invalid/invalid_enum_extension.txt
@@ -1,1 +1,2 @@
 invalid-enum-extension:12:0:12:7:B:"Extending inherited Enum class ""A""":INFERENCE
+invalid-enum-extension:47:0:47:21:IncorrectPastel:"Extending inherited Enum class ""IncorrectColorEnum""":INFERENCE

